components/modal/RegisterModal/index.js

```diff
@@ -36,14 +42,75 @@ export default function RegisterModal() {
     setFormState({ ...formState, success: true });
   };
 
+  /** WIP: If the query includes a code, sign in with google.
+   * Google return URL should also include a role of teacher or student
+   */
+  useEffect(() => {
```

Comment:
Note to self: This huge, and should be split up between facebook & google logins